frontend/packages/knative-plugin/src/utils/knative-topology-utils.ts

```diff
@@ -40,6 +43,45 @@ export enum EdgeType {
   Traffic = 'revision-traffic',
   EventSource = 'event-source-link',
 }
+/**
+ * fetch the parent resource from a resource
+ * @param resource
+ * @param resources
+ */
+export const getParentResource = (
+  resource: K8sResourceKind,
+  resources: K8sResourceKind[],
+): K8sResourceKind => {
+  const parentUids = _.map(resource.metadata.ownerReferences, (owner) => owner.uid);
```

Comment:
nit: for scenarios where it's returning `undefined` , ideally it won't happen but it's good to have checks for `resource.metadata.ownerReferences`